tests/unit/CMakeLists.txt

```diff
@@ -56,6 +56,7 @@ set(UNIT_TESTS_SRCS
   fem/test_assembly_levels.cpp
   fem/test_bilinearform.cpp
   fem/test_blocknonlinearform.cpp
+  fem/test_build_dof_to_arrays.cpp
```

Comment:
Why are there so many existing tests being added here? Were they mistakenly omitted from being built in the past, or was there a reason to exclude them?

Reply:
It looked to me like the were mistakenly omitted. The Makefile build includes them (in `tests/unit`), so I would imagine we want them tested for CMake builds too. I only noticed it because I had a failing CI test for a Makefile build that I wasn't catching locally because I was running `make unit_tests` on a CMake build that was excluding the failing test `.cpp` file.

Reply:
I'm sure some of them were mistakes but I don't know about all of them. 

I often forget about the CMake so the following tests that I added were probably accidentally left out of `CMakeLists.txt`:
```
fem/test_build_dof_to_arrays.cpp
fem/test_doftrans.cpp
fem/test_domain_int.cpp
fem/test_eigs.cpp
fem/test_project_bdr.cpp
fem/test_surf_blf.cpp
```

I believe these two tests were added by @camierjs. John where these intentionally left out of `CMakeLists.txt`? 
```
fem/test_linearform_ext.cpp
fem/test_quadinterpolator.cpp
```

I believe this test was added by @YohannDudouit. Yohann was this intentionally left out of `CMakeLists.txt`?
```
fem/test_sparse_matrix.cpp
```